packages/keyring/src/keyring.ts

```diff
@@ -192,8 +193,20 @@ export class Keyring implements KeyringInstance {
       }
     }
 
-    // FIXME Need to support Ethereum-type derivation paths
-    const derived = keyFromPath(keypairFromSeed[type](seed), path, type);
+    let derived:Keypair;
+
+    if (type === 'ethereum') {
+      const key = HDKeyEth.fromMasterSeed(seed);
+      const child = key.derive(derivePath.substring(1));
+
+      if (child.publicKey && child.privateKey) {
+        derived = { publicKey: child.publicKey, secretKey: child.privateKey };
+      } else {
+        throw new Error('child derivation errored');
+      }
```

Comment:
```suggestion
      assert(child.publicKey && child.privateKey, 'Unable to derive HD key from path');
      
      derived = { publicKey: child.publicKey, secretKey: child.privateKey };
```

Nothin major, just a suggestion.